index.d.ts

```diff
@@ -541,6 +541,27 @@ console.log(stdout);
 export function execaCommand(command: string, options?: Options): ExecaChildProcess;
 export function execaCommand(command: string, options?: Options<null>): ExecaChildProcess<Buffer>;
 
+type TemplateExpression = string | number | Array<string | number>;
+
+type Execa$<EncodingType = string> = {
+	(options: Options): Execa$;
```

Comment:
```suggestion
	(options: Options): Execa$<EncodingType>;
```

We should also add some type tests for it, like:

```ts
expectType<ExecaReturnValue<Buffer>>(await $({encoding: null})({})`unicorns foo`);
expectType<ExecaReturnValue>(await $({encoding: null})({encoding: 'utf8'})`unicorns foo`);
```

Reply:
Included the above suggestion here: https://github.com/sindresorhus/execa/pull/510#discussion_r1109290963

Reply:
Included the above suggestion here: https://github.com/sindresorhus/execa/pull/510#discussion_r1109290963